### ISSUE TITLE
BridgeAdapter support for client and server data store synchronization?

### ISSUE THREAD

**Issue by kristianmandrup** [emberjs/data#868]
I have been trying to do some initial design and experimentation with a bridge adapter which can be set up to store and retrieve data both from a remote (server) data store (via REST) and a local storage on the client.

https://github.com/kristianmandrup/ember-bridge-adapter

However, trying to come up with an initial design for this, I can see that the _ember-data_ API is not quite ready for this approach.

Specifically I see a problem with this section (from `createRecord` in store.js)

```
      adapter = this.adapterForType(type);
      if (adapter && adapter.generateIdForRecord) {
        id = coerceId(adapter.generateIdForRecord(this, record));
        properties.id = id;
      }
    }
```

Since a BridgeAdapter would wraps two adapters, where one generates a local record id and one doesn't (leaving it to the server), this approach just wont work.

As I see it, the problem is that with two different data stores, you might well have two different ID strategies, and which one then becomes the _master_? I would think you could solve this, by having the local storage be master in this scenario (using `generateIdForRecord`)  and then having the BridgeAdapter keep track of which local IDs correspond to which generated server record IDs for each type (model) or something like that? 
Basically an IDs lookup map. What do you think?
